.github/workflows/linux.yml

```
@@ -26,7 +26,7 @@ jobs:
         os: [ubuntu-20.04, ubuntu-22.04]
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```

You need v4 by now.

```suggestion
      - uses: actions/checkout@v4
```

https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/